src/main/java/core/basesyntax/service/impl/CsvReportProvider.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Storage;
+import core.basesyntax.service.ReportProvider;
+import java.util.function.Predicate;
+
+public class CsvReportProvider implements ReportProvider {
+    public static final String REPORT_TITLE = "fruit,quantity";
+    public static final String COLUMN_SEPARATOR = ",";
+    public static final String ROW_SEPARATOR = System.lineSeparator();
+
+    private final Predicate<Integer> predicate = operationValue -> {
+        if (operationValue < 0) {
+            throw new RuntimeException("Operation value cannot be negative");
+        }
+        return true;
+    };
+
+    @Override
+    public String make(Storage storage) {

```

Comment:
we can also have a direct access to the `storage` here instead of passing it as a param